### PR TITLE
Add TS Check

### PR DESCRIPTION
**Pull request by mylesw27**
This PR seeks to add type checking to several of the instructor pages as well as `news_item.js` and `news_items.js`. This PR is part of the larger modernization efforts. As part of that effort, this PR intentionally only adds type checking and fixes any errors identified by TypeScript. Other steps towards modernizing these pages will come in subsequent PRs.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.js [hunk 1]
```diff
@@ -89,31 +90,31 @@ router.post('/', function (req, res, next) {
     res.clearCookie('pl_requested_course_instance_role');
     res.clearCookie('pl_requested_mode');
     res.clearCookie('pl_requested_date');
-    res.cookie('pl_requested_data_changed');
+    res.cookie('pl_requested_data_changed', null);
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.js:93]: Rather than `null`, I'd use a value like `'1'` or `'true'`.

Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.js]: This value has been updated to `'true'` in commit [fd97b5c](https://github.com/PrairieLearn/PrairieLearn/pull/9091/commits/fd97b5c1776985f4dba6a11318040ed5a4790d1f).

Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.js]: Sorry if this wasn't clear, you should make this change for *all* `res.cookie(...)` calls that you changed to take `null` as a second argument.

Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.js]: That makes sense! Sorry, I misunderstood. I have now changed the value for all cookies to be `'true'`.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**